docs/concepts/dht.md

```diff
@@ -1,86 +1,74 @@
 
-- the first list contains peers whose IDs have a different first bit.
-- the second list contains peers whose IDs have first bits identical to its own, but a different second bit
-- ...
-- the m-th list contains peers whose IDs have their first m-1 bits identical, but a different m-th bit
-- ...
+| Have | Want | Don't Want |
+| --- | --- | --- |
+| I can provide this block. | I am looking for this block. | I am not looking for this block. If I am provided this block, I will just discard it. |
 
-The higher m is, the harder it is to find peers that have the same ID up to m bits. The lists of "closest" peers typically remains empty.
-"Close" here is defined as the XOR distance, so the longer the prefix they share, the closer they are.
-Lists also have a maximum of entries (k) — otherwise the first lists would contain half the network, then a fourth of the network, and so on.
+## Re-providing
 
-## How to use DHTs
+If a node announces to the network that it can provide a particular CID, the state of that information is now outside the control of the node. If the node were to drop off the network, there's no way to announce that the CID is no longer available. Take this scenario:
 
-When a peer receives a lookup request, it will either answer with a value if it falls into its own bucket, or answer with the contacting information (IP+port, `peerID`, etc.) of a closer peer. The requesting peer can then send its request to this closer peer. The process goes on until a peer is able to answer it.
-A request for a hash of length n will take at maximum log2(n) steps, or even log2m(n).
+1. `Node A` announces to the network that it can provide `CID X`.
+1. `Node B` makes a record that `Node A` can provide `CID X`.
+1. `Node A` loses its internet connection and can no longer provide anything.
+1. `Node B` isn't aware that `Node A` can no longer provide `CID X`.
+1. `Node C` asks for `CID X`.
+1. `Node B` sends `Node C` to `Node A`.
+1. `Node C` waits for `Node A` to respond until the heat-death of the universe happens, and energy no longer exists.
 
-### Keys and Hashes
+To avoid problem nodes must re-announce which CIDs they can provide. This happens at least every 12 hours. If `Node B` doesn't get a re-announcement from `Node A` that they can still provide `CID X` within a 12 hour period, `Node B` will remove `Node A` from the provider list.
 
-In IPFS's Kademlia DHT, keys are SHA256 hashes. [PeerIDs](https://docs.libp2p.io/concepts/peer-id/) are those of [libp2p](https://libp2p.io/), the networking library used by IPFS.
+## Dual DHT
 
-We use the DHT to look up two types of objects, both represented by SHA256 hashes:
+IPFS nodes run two DHTs: one for the public internet WAN, and one for their local network LAN.
```

Comment:
maybe "participate in"?

It might also be useful to make the distinction above between querying a DHT and acting as a DHT server. Nodes will participate in the DHT from their LAN, and will store some of that generated metadata, but only expect that DHT to be used when the LAN is disconnected. Nodes will only store part of the public DHT when they are externally reachable (not behind a NAT).

Reply:
Addressed in 7af0c8f